src/pagination.rs

```diff
@@ -0,0 +1,217 @@
+use crate::{Button, GlobalClose, Variant};
+use yew::prelude::*;
+
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct Props {
+    #[prop_or_default]
+    pub total_entries: Option<i32>,
+    #[prop_or_default]
+    pub offset: i32,
+    #[prop_or(vec![10,20,30])]
+    pub entries_per_page_choices: Vec<i32>,
+    #[prop_or(20)]
+    pub selected_choice: i32,
```

Comment:
Might make sense to keep the name similar to what's in the upstream TypeScript definition. Nomenclature is page. I think that still makes sense here.